music21/alpha/analysis/ornamentRecognizer.py

```diff
@@ -150,24 +146,243 @@ def recognize(self) -> Union[bool, expressions.Trill]:
         trill.size = distance
         return trill
 
+class TurnRecognizer(OrnamentRecognizer):
+    def __init__(self, ):
+        self.acceptableInterval = 3
+        self.minimumLengthForNachschlag = 6
+
+    def recognize(self, busyNotes, simpleNotes=None)\
+            -> Union[bool, expressions.Turn, expressions.InvertedTurn]:
+        '''
+        Tries to identify the busy notes as a turn or inverted turn.
+
+        When simple notes is provided, tries to identify busy notes
+        as the turn shortened by simple notes.
+        Currently only supports one simple note in simple notes.
+
+        Turns and inverted turns have four notes separated by m2, M2, A2.
+
+        Turns:
+        start above base note
+        go down to base note,
+        go down again,
+        and go back up to base note
+
+        Inverted Turns:
+        start below base note
+        go up to base note,
+        go up again,
+        and go back down to base note
+
+        When going up or down, must go to the adjacent note name,
+        so A goes down to G, G#, G flat, G##, etc
+
+        Returns: False if not possible or the Turn/Inverted Turn Expression
+        '''
+        # number of notes/ duration of notes ok
+        if len(busyNotes) != 4:
+            return False
+
+        if simpleNotes:
+            eps = .1
+            totalBusyNotesDuration = 0
+            for n in busyNotes:
+                totalBusyNotesDuration += n.duration.quarterLength
+            if abs(simpleNotes[0].duration.quarterLength - totalBusyNotesDuration) > eps:
+                return False
+
+        # pitches ok
+        if busyNotes[1].pitch.midi != busyNotes[3].pitch.midi:
+            return False
+        if simpleNotes and simpleNotes[0].pitch.midi != busyNotes[1].pitch.midi:
+            return False
+
+        # intervals ok
+        acceptableIntervals = [interval.Interval("M2"), interval.Interval("M-2"),
```

Comment:
This creates 6 new interval objects for each possible set of notes.  Run `%timeit` on creating an interval object and see if this might be a problem or not.  Is there a more efficient way of doing this?

Reply:
Couldn't figure out how to run timeit successfully, but I realized this is inefficient. Code is changed to initialize acceptable intervals only when fixer instance initialized. Is this enough of an improvement or were you thinking something even better?